setup.py

```diff
@@ -40,6 +42,9 @@
     'arbor>=0.7',
 ]
 
+if sys.version_info < (3, 8):
```

Comment:
I wonder if we should really do a hardcore requirement?
It's very possible it would work in older versions.

Reply:
Yeah we don't have to do it.

Python 3.7 will work but it is likely that users soon will start having compatibility errors due to python3.7 drop of support. Those errors may or may not clearly state that the issue is raised because of the python version and thus confuse the users.

I guess it's taste at the end of the day. I can revert that single commit. Let me know.